runtime/lua/vim/inspect.lua

```diff
@@ -289,7 +289,7 @@ function Inspector:putValue(v)
   if tv == 'string' then
     self:puts(smartQuote(escape(v)))
   elseif tv == 'number' or tv == 'boolean' or tv == 'nil' or
-         tv == 'cdata' or tv == 'ctype' then
+         tv == 'cdata' or tv == 'ctype' or (vim and v == vim.NIL) then
```

Comment:
why do we need to check for `vim` first here?

Reply:
because the file is also used in `screen.lua` in the tests. Though the goal is that the tests will run in nvim-lua (rather than the separate lua-client) eventually, and then this check won't be needed.